WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV0.swift

```diff
@@ -72,9 +72,37 @@ class ConversationsAPIV0: ConversationsAPI, VersionedAPI {
         assertionFailure("not implemented! use getLegacyConversationIdentifiers() instead")
         throw ConversationsAPIError.notImplemented
     }
+
+    func getConversations(for identifiers: [QualifiedID]) async throws -> ConversationList {
+        let parameters = GetConversationsParametersV0(qualifiedIdentifiers: identifiers)
+        let body = try JSONEncoder.defaultEncoder.encode(parameters)
+        let resourcePath = "/conversations/list/v2"
+
+        let request = HTTPRequest(
+            path: resourcePath,
+            method: .post,
+            body: body
+        )
+        let response = try await self.httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: 200, type: QualifiedConversationListV0.self)
+            .failure(code: 400, error: ConversationsAPIError.invalidBody)
+            .parse(response)
+    }
 }
 
-// MARK: -
+// MARK: Encodables
+
+struct GetConversationsParametersV0: Encodable {
```

Comment:
can it be private ?

Reply:
No, it's also used in V2, because only the path has changed but the enconding/decoding stayed the same. To avoid not necessary duplication I use the same structs.

I think the root is that we decided to use inheritance to make this dance around api versions instead of other solutions. This way we need to implement the whole function again even if a tiny bit changes:
1. create request
2. execute request
3. parse response

Any idea how to solve this case using inheritance in another way? cc @johnxnguyen @typfel

Reply:
Thanks for the suggestion John. I would like to do this idea in a separate PR. @netbe is your question clarified so far?

Reply:
yes